### ISSUE TITLE
NUnit2007 shouldn't trigger for generic types e.g. typeof(T)

### ISSUE THREAD

**Issue by mikkelbu** [nunit/nunit.analyzers#896]
I noticed the following error in the docs build of snippet (included below). We should not trigger on cases like this as the actual value is not a constant whereas the expected value is.

![Image](https://github.com/user-attachments/assets/87510d48-6fda-4a63-a2a0-a569d606fad5)

https://github.com/nunit/docs/pull/1052/files?show-viewed-files=true&file-filters%5B%5D=#annotation_35790100039

```csharp
    public class MyExplicitlyTypedTests
    {
        [TestCaseSource(nameof(ExplicitTypeArgsTestCases))]
        public void ExplicitTypeArgs<T>(T input)
        {
            Assert.That(typeof(T), Is.EqualTo(typeof(long)));
        }

        private static IEnumerable<TestCaseData> ExplicitTypeArgsTestCases()
        {
            yield return new TestCaseData(2) { TypeArgs = new[] { typeof(long) } };
            yield return new TestCaseData(2L) { TypeArgs = new[] { typeof(long) } };
        }
    }
```
